addons/survey/static/src/views/survey_views.js

```diff
@@ -0,0 +1,81 @@
+/** @odoo-module **/
+
+import { registry } from '@web/core/registry';
+import { ListRenderer } from "@web/views/list/list_renderer";
+import { KanbanRenderer } from "@web/views/kanban/kanban_renderer";
+import { listView } from '@web/views/list/list_view';
+import { kanbanView } from '@web/views/kanban/kanban_view';
+import { useService } from "@web/core/utils/hooks";
+import { patch } from "@web/core/utils/patch";
+
+const { useEffect, useRef } = owl;
+
+const SurveySampleMixin = {
+    useSurveyLoadSampleHook(selector) {
+        const rootRef = useRef("root");
+        useEffect(
+            (elems) => {
+                if (!elems || !elems.length) {
+                    return;
+                }
+                const handler = (ev) => {
+                    const surveyMethod = ev.currentTarget.closest('.o_survey_sample_container').getAttribute('action');
+                    this.loadSample(surveyMethod);
+                }
+                for (const elem of [...elems]) {
```

Comment:
Actually I was wrong, we don't need to unpack elems into a list.
Same in the cleanup function
```suggestion
                for (const elem of elems) {
```